docs/build-macos.md

```diff
@@ -0,0 +1,52 @@
+macOS builds can be created using:
+
+- Clang, using Meson buildsystem and Homebrew for installing dependencies
+  (*recommended*)
+- GCC, using Meson buildsystem and Homebrew for installing dependencies.
+- Clang or GCC, using Meson buildsystem and MacPorts for dependencies.
```

Comment:
MacOS builds can be created using the Meson build system with your choice of:
 - Clang or GCC for compiling, and
 - Brew or MacPorts for provision of dependencies.

We recommend Brew and Clang, and specifically the latter because only Apple's Clang can link with Apple's Core SDK libraries.